### PR TITLE
1315 add a comment to the preview tab

### PR DESCRIPTION
**Pull request by zimboflyman**
### What is the context of this PR?
On the preview page:
Previous right hand section 'Properties' - content has been removed
Replaced by A new comments area.
Here we are able to create new comments in the text area.
newest comments should be displayed at the top of the list
Currently unable to reply, delete or edit comments as per this ticket

### How to review 
Create a new page on a questionnaire.
Click on 'Preview' tab
Add a comment in the right hand comments Txt area
This should be displayed under the text area with the initials of you user name in the avatar area and today's time along with today
All should be as per the designs on the ticket
Delete and reply buttons should be not be visible
You should be able to add multiple comments under multiple users.
Newest comments shown at top of list.

ensure all tests pass and they cover all options above

### CODE REVIEW

**Comment by zimboflyman** [comment]
[Add a comment to the preview tab (10)](https://trello.com/c/R3OxLW3j/1315-add-a-comment-to-the-preview-tab-10)

File: eq-author/src/App/page/Preview/CommentsPanel/index.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  height: 22px;
+`;
+
+const DateField = styled("span")`
+  font-weight: normal;
+  font-size: 0.8em;
+  color: ${colors.grey};
+`;
+
+const DeleteButtonIcon = styled.button`
```
Comment (SamGodwin2) [inline eq-author/src/App/page/Preview/CommentsPanel/index.js]: We already have a component for this sorta thing, think its called an IconButton.

File: eq-author-api/schema/resolvers/pages/index.js [hunk 2]
```diff
@@ -19,6 +30,17 @@ Resolvers.Page = {
   },
 };
 
+Resolvers.Comment = {
+  commentBody: ({ commentText }) => commentText,
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/resolvers/pages/index.js]: thanks Sam - I've changed the queries and tests to reflect

File: eq-author-api/schema/resolvers/pages/questionPage.js [hunk 3]
```diff
@@ -46,6 +46,14 @@ Resolvers.QuestionPage = {
       true,
       ROUTING_ANSWER_TYPES
     ),
+  comments: async ({ id }, args, ctx) => {
+    const questionnaireId = ctx.questionnaire.id;
+    const questionnareComments = await getCommentsForQuestionnaire(
+      questionnaireId
+    );
+    // console.log(questionnareComments.comments);
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/resolvers/pages/questionPage.js]: woops!!

File: eq-author/src/App/page/Preview/CommentsPanel/index.js (hunk 1 above)
Comment (zimboflyman) [reply eq-author/src/App/page/Preview/CommentsPanel/index.js]: removed both the delete and reply buttons until those functions are added

File: eq-author-api/schema/tests/page.test.js [hunk 4]
```diff
@@ -78,8 +79,9 @@ describe("page", () => {
         position: 0,
       });
 
-      const readPage = await queryPage(ctx, createdPage.id);
+      // console.log(createdPage);
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/tests/page.test.js]: Woops again!

File: eq-author-api/schema/tests/comments.test.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    await deleteQuestionnaire(ctx, questionnaire.id);
+    questionnaire = null;
+  });
+
+  beforeEach(async () => {
+    ctx = await buildContext({
+      metadata: [{}],
+      sections: [
+        {
+          page: [{}],
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/tests/comments.test.js]: understood!  thanks

File: eq-author-api/schema/tests/comments.test.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    });
+  });
+
+  it("should query a comment and resolve comment fields", async () => {
+    const comment = await queryComments(ctx, {
+      pageId: createdPage.id,
+    });
+    expect(comment).toMatchObject({
+      id: expect.any(String),
+      comments: expect.any(Array),
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/tests/comments.test.js]: Changed the description of this test to show that we are creating an empty comment array on new questionnaire.

File: eq-author-api/schema/resolvers/pages/questionPage.js [hunk 7]
```diff
@@ -46,6 +46,14 @@ Resolvers.QuestionPage = {
       true,
       ROUTING_ANSWER_TYPES
     ),
+  comments: async ({ id }, args, ctx) => {
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/schema/resolvers/pages/questionPage.js]: Agreed with Sarah that we will add comments to calculated summary page,
will make a new ticket for Question intro page
no comments on section intro page

File: eq-author-api/schema/resolvers/base.js [hunk 8]
```diff
@@ -221,6 +222,7 @@ const Resolvers = {
         ...args.input,
         createdBy: ctx.user.id,
       });
+      await createComments(questionnaire.id);
```
Comment (SamGodwin2) [inline eq-author-api/schema/resolvers/base.js]: Gonna need to add a migration to retroactively add comments objects to previous questionnaires.

File: eq-author/src/components/CommentsPanel/index.js [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+import CustomPropTypes from "custom-prop-types";
+
+import COMMENT_QUERY from "./commentsQuery.graphql";
+import COMMENT_ADD from "./createNewComment.graphql";
+
+import styled from "styled-components";
+import ScrollPane from "components/ScrollPane";
+import { colors } from "constants/theme";
+
+import TextArea from "components/Forms/TextArea";
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.js]: Need to ask Sarah if this needs to be a rich text editor instead.

File: eq-author/src/components/CommentsPanel/commentsQuery.graphql [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+          id
+          name
+          picture
+          email
+          displayName
+        }
+      }
+    }
+  }
+}
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/commentsQuery.graphql]: I think there should be a newline at the end of this file.

File: eq-author/src/components/CommentsPanel/createNewComment.graphql [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+      user {
+        id
+        name
+        picture
+        email
+        displayName
+      }
+    }
+  }
+}
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/createNewComment.graphql]: I think there should be a new line at the end of this file.

File: eq-author/src/components/CommentsPanel/index.js [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+          },
+        },
+      });
+      setComment("");
+    }
+    event.preventDefault();
+  };
+
+  const getInitials = name => {
+    const initials = name.replace(/[^a-zA-Z- ]/g, "").match(/\b\w/g);
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.js]: The name `Jimbo Billy Bob The First` causes this to throw an error.

File: eq-author/src/components/CommentsPanel/index.js [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+    return initials.join("").toUpperCase();
+  };
+
+  //make sure qraphQL has returned before rendering
+  if (loading) {
+    return <Loading height="100%">Comments loading…</Loading>;
+  }
+  if (error) {
+    return <Error>Oops! Something went wrong</Error>;
+    // return <p>{error.message}</p>;
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.js]: No need for this comment.

File: eq-author/src/components/CommentsPanel/index.js [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+  padding: 0.5em 1em;
+  margin-bottom: 5px;
+  border: 1px solid ${colors.lightGrey};
+  background-color: ${colors.lighterGrey};
+`;
+
+const CommentHeaderContainer = styled(Field)`
+  display: flex;
+  justify-content: flex-start;
+  -webkit-align-items: start;
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/index.js]: Is there a reason why you use the newer version of `align-items` within `ShowComment` but the older version within `CommentHeaderContainer`? Should the old and new version be included in each, respectively, as well as what already exists?

File: eq-author/src/components/CommentsPanel/index.js [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+  height: 22px;
+`;
+
+const DateField = styled("span")`
+  font-weight: normal;
+  font-size: 0.8em;
+  color: ${colors.grey};
+`;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/index.js]: Why is this here?

File: eq-author/src/components/CommentsPanel/index.test.js [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+  let props, pageId, queryWasCalled, mocks;
+
+  // this is just a little hack to silence a warning that we'll get until we
+  // upgrade to 16.9: https://github.com/facebook/react/pull/14853
+  // https://github.com/testing-library/react-testing-library#suppressing-unnecessary-warnings-on-react-dom-168
+  actSilenceWarning();
+
+  beforeEach(() => {
+    queryWasCalled = false;
+    props = {
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.test.js]: No need for this props object and no need to pass the props into any of the components below.

File: eq-author/src/components/CommentsPanel/index.test.js [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+  });
+
+  /////////////////////////////////////////////////////////////////////////
+
+  it("should render comments txt area", async () => {
+    const { getByTestId } = render(<CommentsPanel {...props} />, {
+      route: "/q/Q1/page/P1",
+      urlParamMatcher: "/q/:questionnaireId/page/:pageId",
+      mocks,
+    });
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.test.js]: We render the same thing in every test, could we extract it?

File: eq-author/src/components/CommentsPanel/index.js (hunk 13 above)
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/index.js]: Should the comment be removed, or line 143 be replaced with the comment?

File: eq-author/src/components/buttons/Button/__snapshots__/index.test.js.snap [hunk 18]
```diff
@@ -277,6 +277,215 @@ exports[`components/Button renders according to variant 4`] = `
 </button>
 `;
 
+exports[`components/Button renders according to variant 5`] = `
```
Comment (SamGodwin2) [inline eq-author/src/components/buttons/Button/__snapshots__/index.test.js.snap]: Whats a greyed button? Doesn't seem to be used anywhere?

File: eq-author/src/components/buttons/Button/index.js [hunk 19]
```diff
@@ -97,6 +115,8 @@ const Button = styled.button`
   ${props => props.variant === "secondary" && secondaryButton};
   ${props => props.variant === "tertiary" && tertiaryButton};
   ${props => props.variant === "tertiary-light" && tertiaryLightButton};
+  ${props => props.variant === "greyed" && greyedButton};
```
Comment (SamGodwin2) [inline eq-author/src/components/buttons/Button/index.js]: greyed?

File: eq-author/src/components/CommentsPanel/index.js [hunk 20]
```diff
@@ -0,0 +1,202 @@
...
+    }
+    event.preventDefault();
+  };
+
+  const getInitials = name => {
+    const initials = name.replace(/[^a-zA-Z- ]/g, "").match(/\b\w/g);
+    return initials.join("").toUpperCase();
+  };
+
+  //make sure qraphQL has returned before rendering
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/index.js]: I don't think this comment needs to be here, we do this quite a lot throughout the application and what the code does is fairly self explanatory.

File: eq-author/src/components/buttons/Button/__snapshots__/index.test.js.snap (hunk 18 above)
Comment (zimboflyman) [reply eq-author/src/components/buttons/Button/__snapshots__/index.test.js.snap]: was a new variant for the reply button - now removed until we implement reply

File: eq-author/src/components/CommentsPanel/index.js (hunk 13 above)
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.js]: removed

File: eq-author-api/tests/utils/contextBuilder/comments/deleteComment.js [hunk 21]
```diff
@@ -0,0 +1,25 @@
+const executeQuery = require("../../executeQuery");
+
+const deleteQuestionnaireMutation = `
+  mutation DeleteQuestionnaire($input: DeleteCommentInput!) {
+    deleteComment(input: $input) {
+      id
+    }
+  }
+`;
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/tests/utils/contextBuilder/comments/deleteComment.js]: this file not needed until we implement delete funct - now deleted!

File: eq-author-api/tests/utils/contextBuilder/comments/queryComment.js [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+            commentText
+            user {
+                id
+                name
+                picture
+                email
+                displayName
+            }
+            createdTime
+            replies {
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/tests/utils/contextBuilder/comments/queryComment.js]: removed

File: eq-author-api/tests/utils/contextBuilder/comments/deleteComment.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
+const executeQuery = require("../../executeQuery");
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/tests/utils/contextBuilder/comments/deleteComment.js]: file deleted

File: eq-author-api/tests/utils/contextBuilder/comments/createComment.js [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+        user {
+            id
+            name
+            picture
+            email
+            displayName
+        }
+        replies {
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/tests/utils/contextBuilder/comments/createComment.js]: removed

File: eq-author/src/components/CommentsPanel/commentsQuery.graphql [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+      commentText
+      user {
+        id
+        name
+        picture
+        email
+        displayName
+      }
+      createdTime
+      replies {
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/commentsQuery.graphql]: done

File: eq-author/src/components/CommentsPanel/commentsQuery.graphql (hunk 10 above)
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/commentsQuery.graphql]: now removed as not implementing delete comments yet

File: eq-author-api/tests/utils/contextBuilder/comments/index.js [hunk 26]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  ...require("./createComment"),
+  ...require("./queryComment"),
+  ...require("./deleteComment"),
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author-api/tests/utils/contextBuilder/comments/index.js]: done

File: eq-author/src/components/CommentsPanel/index.js (hunk 12 above)
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.js]: fix added for empty names and maximum of first 3 initials so as not to overfloww the avatar circle

File: eq-author/src/components/CommentsPanel/index.test.js (hunk 17 above)
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.test.js]: done

File: eq-author/src/components/CommentsPanel/commentsQuery.graphql (hunk 10 above)
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/commentsQuery.graphql]: Both have lines! - good to know

File: eq-author-api/schema/resolvers/base.js (hunk 8 above)
Comment (zimboflyman) [reply eq-author-api/schema/resolvers/base.js]: ahh, good point! - can this be a seperate ticket or must go in with this PR?

File: eq-author/src/components/CommentsPanel/index.js [hunk 27]
```diff
@@ -0,0 +1,202 @@
...
+                  <AvatarOuter>
+                    <AvatarInner>{getInitials(item.user.name)}</AvatarInner>
+                  </AvatarOuter>
+                </AvatarWrapper>
+                <NameWrapper>
+                  <FlexLabel>{item.user.name}</FlexLabel>
+                  <DateField>{moment(item.createdTime).calendar()}</DateField>
+                </NameWrapper>
+              </CommentHeaderContainer>
+              <ShowComment data-test="XYZ">{item.commentText}</ShowComment>
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.js]: Do we need this test attrib and if so could we make it a bit more descriptive?

File: eq-author/src/components/CommentsPanel/index.test.js [hunk 28]
```diff
@@ -0,0 +1,197 @@
...
+    const { getByText } = render(<CommentsPanel />, { ...props });
+
+    await flushPromises();
+    const addCommentButton = getByText("Add");
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(addCommentButton).toHaveAttribute("disabled");
+  });
+
+  it("should enable add button when comment txt area is hs content", async () => {
```
Comment (SamGodwin2) [inline eq-author/src/components/CommentsPanel/index.test.js]:
```suggestion
  it("should enable add button when comment txt area has content", async () => {
```

File: eq-author/src/components/CommentsPanel/index.js [hunk 29]
```diff
@@ -0,0 +1,202 @@
...
+          <React.Fragment key={item.id}>
+            <CommentSection>
+              <CommentHeaderContainer>
+                <AvatarWrapper>
+                  <AvatarOuter>
+                    <AvatarInner>{getInitials(item.user.name)}</AvatarInner>
+                  </AvatarOuter>
+                </AvatarWrapper>
+                <NameWrapper>
+                  <FlexLabel>{item.user.name}</FlexLabel>
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.js]: coolio - good to know

File: eq-author/src/components/CommentsPanel/index.js (hunk 27 above)
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.js]: not used so removed

File: eq-author/src/components/CommentsPanel/index.js [hunk 30]
```diff
@@ -0,0 +1,202 @@
...
+              data-test="btn-add-comment"
+            >
+              Add
+            </Button>
+          </ButtonGroup>
+        </CommentSection>
+
+        {commentsArray.map(item => (
+          <React.Fragment key={item.id}>
+            <CommentSection>
```
[the remark this answers is not in the document]
Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.js]: nope - not needed, so removed and moved the key

File: eq-author-api/schema/resolvers/base.js (hunk 8 above)
Comment (zimboflyman) [reply eq-author-api/schema/resolvers/base.js]: done

File: eq-author-api/schema/typeDefs.js [hunk 31]
```diff
@@ -24,6 +24,11 @@ type QuestionnaireInfo {
   totalSectionCount: Int!
 }
 
+type PublishDetails {
+  surveyId: String
+  formType: JSON
+}
+
```
Comment (MattElcock) [inline eq-author-api/schema/typeDefs.js]: This shouldn't be here; looks like something I should have removed from one of my PRs accidentally got forgotten about by myself. If you could remove it, that would be grand.

File: eq-author-api/schema/typeDefs.js [hunk 32]
```diff
@@ -56,6 +61,7 @@ type Questionnaire {
   permission: Permission!
   isPublic: Boolean!
   publishStatus: PublishStatus!
+  publishDetails: PublishDetails
```
Comment (MattElcock) [inline eq-author-api/schema/typeDefs.js]: This should go, too, for the same reason as above.

File: eq-author/src/App/page/Preview/CalculatedSummaryPreview.test.js [hunk 33]
```diff
@@ -1,13 +1,28 @@
...
+
   beforeEach(() => {
+    me = {
+      id: "123",
+      displayName: "Raymond Holt",
+      email: "RaymondHolt@TheNineNine.com",
+      picture: "http://img.com/avatar.jpg",
+      admin: true,
+    };
+
```
Comment (MattElcock) [inline eq-author/src/App/page/Preview/CalculatedSummaryPreview.test.js]: Love it. 😜

File: eq-author/src/App/publish/PublishPage.js [hunk 34]
```diff
@@ -0,0 +1,161 @@
+import React, { useState, useEffect } from "react";
```
Comment (MattElcock) [inline eq-author/src/App/publish/PublishPage.js]: Why is the Publish Page here as if it's a new feature in this PR? It was merged in the other day.

File: eq-author/src/App/publish/PublishPage.test.js [hunk 35]
```diff
@@ -0,0 +1,99 @@
+import React from "react";
```
Comment (MattElcock) [inline eq-author/src/App/publish/PublishPage.test.js]: Same as above, this was merged in the other day. Looks like a rebase has gone a bit weird.

File: eq-author/src/App/publish/index.js [hunk 36]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
```
Comment (MattElcock) [inline eq-author/src/App/publish/index.js]: Same as above.

File: eq-author/src/components/EditorLayout/Header/index.js [hunk 37]
```diff
@@ -13,6 +13,7 @@ import { withMe } from "App/MeContext";
 
 import Button from "components/buttons/Button";
 import LinkButton from "components/buttons/Button/LinkButton";
+import RouteButton from "components/buttons/Button/RouteButton";
```
Comment (MattElcock) [inline eq-author/src/components/EditorLayout/Header/index.js]: Same as above, this should have been merged in from a previously merged PR.

File: eq-author/src/components/EditorLayout/Header/index.js [hunk 38]
```diff
@@ -23,11 +24,12 @@ import viewIcon from "./icon-view.svg?inline";
 import settingsIcon from "./icon-cog.svg?inline";
 import publishIcon from "./icon-publish.svg?inline";
 import SharingModal from "./SharingModal";
-import PublishModal from "./PublishModal";
 import PageTitle from "./PageTitle";
 import UpdateQuestionnaireSettingsModal from "./UpdateQuestionnaireSettingsModal";
 import SavingIndicator from "./SavingIndicator";
 
+import { buildPublishPath } from "utils/UrlUtils";
```
Comment (MattElcock) [inline eq-author/src/components/EditorLayout/Header/index.js]: Same as above ^

File: eq-author/src/components/EditorLayout/Header/index.js [hunk 39]
```diff
@@ -135,15 +132,18 @@ export class UnconnectedHeader extends React.Component {
...
-                      disabled={questionnaire.totalErrorCount > 0}
+                      disabled={
+                        questionnaire.totalErrorCount > 0 ||
+                        this.props.title === "Publish"
+                      }
+                      data-test="btn-publish"
                     >
                       <IconText icon={publishIcon}>Publish</IconText>
-                    </Button>
+                    </RouteButton>
```
Comment (MattElcock) [inline eq-author/src/components/EditorLayout/Header/index.js]: Same as above ^

File: eq-author/src/components/EditorLayout/Header/index.test.js [hunk 40]
```diff
@@ -116,6 +119,14 @@ describe("Header", () => {
...
       expect(publishSurveyButton).not.toHaveAttribute("disabled");
     });
+
+    it("should disable the publish survey button when the user is currently on the publish page", () => {
+      props.title = "Publish";
+      const { getByTestId } = renderWithContext(<Header {...props} />);
+
+      const publishSurveyButton = getByTestId("btn-publish");
+      expect(publishSurveyButton).toHaveAttribute("disabled");
+    });
```
Comment (MattElcock) [inline eq-author/src/components/EditorLayout/Header/index.test.js]: Same as above

File: eq-author/src/components/CommentsPanel/index.test.js [hunk 41]
```diff
@@ -0,0 +1,197 @@
...
+    expect(queryWasCalled).toBeTruthy();
+    expect(getByText("This is a test ADD comment")).toBeTruthy();
+  });
+
+  it("should render loading state", () => {
+    const { getByTestId } = render(<CommentsPanel />, { ...props });
+    expect(getByTestId("loading")).toBeTruthy();
+  });
+
+  it("should render error state", async () => {
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/index.test.js]:
```suggestion
  it("should render error state on failed query", async () => {
```

File: eq-author/src/components/buttons/Button/RouteButton.js [hunk 42]
```diff
@@ -5,7 +5,17 @@ import { Link } from "react-router-dom";
...
+
+const propWhitelist = [
+  "to",
+  "title",
+  "id",
+  "className",
+  "children",
+  "disabled",
+  "data-test",
+];
```
Comment (MattElcock) [inline eq-author/src/components/buttons/Button/RouteButton.js]: Same as above reg. the publish stuff.

File: eq-author/src/components/buttons/Button/RouteButton.js [hunk 43]
```diff
@@ -16,7 +26,15 @@ class RouteButton extends React.Component {
...
-    const { children, to, ...otherProps } = this.props;
+    const { children, to, disabled, ...otherProps } = this.props;
+
+    if (disabled) {
+      return (
+        <DisabledLink disabled {...otherProps}>
+          {children}
+        </DisabledLink>
+      );
+    }
```
Comment (MattElcock) [inline eq-author/src/components/buttons/Button/RouteButton.js]: Same as above

File: eq-author/src/utils/UrlUtils.js [hunk 44]
```diff
@@ -89,6 +90,13 @@ export const buildHistoryPath = ({ questionnaireId }) => {
   });
 };
 
+export const buildPublishPath = ({ questionnaireId }) => {
+  return generatePath(Routes.QUESTIONNAIRE)({
+    questionnaireId,
+    entityName: PUBLISH,
+  });
+};
+
```
Comment (MattElcock) [inline eq-author/src/utils/UrlUtils.js]: Same as above

File: eq-author/src/utils/UrlUtils.js [hunk 45]
```diff
@@ -8,6 +8,7 @@ import {
   INTRODUCTION,
   METADATA,
   HISTORY,
+  PUBLISH,
```
Comment (MattElcock) [inline eq-author/src/utils/UrlUtils.js]: Same as above

File: eq-author-api/schema/typeDefs.js (hunk 31 above)
Comment (zimboflyman) [reply eq-author-api/schema/typeDefs.js]: Somehow one of your commits has ended up in this - maybe during a rebase?
I have reverted your commit and that looks to have addressed all the comments below...

File: eq-author/src/components/CommentsPanel/index.js [hunk 46]
```diff
@@ -0,0 +1,200 @@
...
+  padding: 0.7em 1em;
+  border-bottom: 1px solid ${colors.lightMediumGrey};
+`;
+
+const ShowComment = styled.div`
+  padding: 0.5em 1em;
+  margin-bottom: 5px;
+  border: 1px solid ${colors.lightGrey};
+  background-color: ${colors.lighterGrey};
+`;
```
Comment (MattElcock) [inline eq-author/src/components/CommentsPanel/index.js]: Can you add `word-break: break-word;` to this class, please? When you make a continuous word with no spaces it'll go outside of the box and extend the page.

Comment (zimboflyman) [reply eq-author/src/components/CommentsPanel/index.js]: ooooh - well spotted!
